code/script.js

```diff
@@ -222,104 +229,129 @@ const generateBoard = () => {
+    } else {
+      alert(`No, the figures secret talent is not ${value}! Remove all that are ${value}`);
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value));
     }
-  } else if (category === 'other') {
-    // Similar to the one above
   } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(`Yes, it is ${category} ${value}! Keep all that are ${category} ${value}`);
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value);
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(`No, it is not ${category} ${value}! Remove all that are ${category} ${value}`);
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value);
     }
   }
-
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
   // Invoke a function to redraw the board with the remaining people.
+  generateBoard()
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  //Here I store the interaction from the player in a variable
+  let playerGuess = confirm(`Do you wish to guess on ${personToConfirm}?`);
+  // If the player wants to guess, invoke the checkMyGuess function. 
+  //I pass playerGuess as param, if the player confirmed yes in the pop-up playerGuess = true and the codeblock runs
+  if (playerGuess){
+    checkMyGuess(personToConfirm);
+  }
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
   // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
+  if (personToCheck === secret.name) {
+  // 2. Set a Message to show in the win or lose section accordingly  
+    winOrLoseText.innerHTML= `Wihuuuu that is correct, the secret figure is ${personToCheck}`;
+  } else {
+    winOrLoseText.innerHTML= `Sorry, but your guess ${personToCheck} is not correct. The secret figure is ${secret.name}`;
+    secretImageAtGuess.innerHTML= `<img src=${secret.img} alt=${secret.name}>`;
```

Comment:
Nice feature with the picture of the secret person in the win-and-loose-section.